### PR TITLE
Moving scalar and field tests (for #1106)

### PR DESCRIPTION
**Pull request by TeranIvy**
Replaces the failed PR #1118.

### CODE REVIEW

File: src/psyclone/tests/domain/lfric/lfric_scalar_codegen_test.py [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+#         I. Kavcic and A. Coughtrie, Met Office;
+#         C. M. Maynard, Met Office/University of Reading;
+#         J. Henrichs, Bureau of Meteorology.
+
+'''
+Module containing pytest tests for PSy-layer code generation for the
+LFRic scalar arguments.
+'''
+
+# Imports
```
Comment (arporter) [inline src/psyclone/tests/domain/lfric/lfric_scalar_codegen_test.py:44]: Please rm this line.

File: src/psyclone/tests/domain/lfric/lfric_scalar_codegen_test.py [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+
+# Constants
+BASE_PATH = os.path.join(
+    os.path.dirname(os.path.dirname(os.path.dirname(
+        os.path.abspath(__file__)))),
+    "test_files", "dynamo0p3")
+TEST_API = "dynamo0.3"
+
+
+# Tests for PSy-layer code generation for scalar arguments
```
Comment (arporter) [inline src/psyclone/tests/domain/lfric/lfric_scalar_codegen_test.py:59]: This is already said in the docstring so please remove.

File: src/psyclone/tests/domain/lfric/lfric_scalar_mdata_test.py [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+#         I. Kavcic and A. Coughtrie, Met Office;
+#         C. M. Maynard, Met Office/University of Reading;
+#         J. Henrichs, Bureau of Meteorology.
+
+'''
+Module containing pytest tests for the general LFRic scalar arguments
+functionality (e.g. metadata, parsing, invoke calls).
+'''
+
+# Imports
```
Comment (arporter) [inline src/psyclone/tests/domain/lfric/lfric_scalar_mdata_test.py:44]: Please rm this line.

File: src/psyclone/tests/domain/lfric/lfric_scalar_mdata_test.py [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+from psyclone.errors import InternalError, GenerationError
+
+# Constants
+BASE_PATH = os.path.join(
+    os.path.dirname(os.path.dirname(os.path.dirname(
+        os.path.abspath(__file__)))),
+    "test_files", "dynamo0p3")
+TEST_API = "dynamo0.3"
+
+# General scalar checks (argument type, data type, etc)
```
Comment (arporter) [inline src/psyclone/tests/domain/lfric/lfric_scalar_mdata_test.py:66]: Please rm this line (plus some blank ones).

File: src/psyclone/tests/dynamo0p3_test.py [hunk 5]
```diff
@@ -81,7 +81,7 @@ def setup():
     Config.get().api = "dynamo0.3"
 
 
-# tests
+# Tests
```
Comment (arporter) [inline src/psyclone/tests/dynamo0p3_test.py:84]: Please rm this line.

**Review by arporter — CHANGES_REQUESTED** [review]
Looks good, it's really nice to see those file size coming down to something less embarrassing.
There are just a few superfluous comments that I couldn't bring myself to ignore. Once those are removed, this is ready to go.
All tests still OK with compilation. Total number of tests is the same as on master.

File: src/psyclone/tests/domain/lfric/lfric_scalar_codegen_test.py (hunk 1 above)
Comment (TeranIvy) [reply src/psyclone/tests/domain/lfric/lfric_scalar_codegen_test.py:44]: Done.

File: src/psyclone/tests/domain/lfric/lfric_scalar_codegen_test.py (hunk 2 above)
Comment (TeranIvy) [reply src/psyclone/tests/domain/lfric/lfric_scalar_codegen_test.py:59]: Yep.

File: src/psyclone/tests/domain/lfric/lfric_scalar_mdata_test.py (hunk 3 above)
Comment (TeranIvy) [reply src/psyclone/tests/domain/lfric/lfric_scalar_mdata_test.py:44]: Done.

File: src/psyclone/tests/domain/lfric/lfric_scalar_mdata_test.py (hunk 4 above)
Comment (TeranIvy) [reply src/psyclone/tests/domain/lfric/lfric_scalar_mdata_test.py:66]: Done.

**Review by arporter — APPROVED** [review]
Requested changes have been made.
All tests and examples still OK.
All docs still build OK.
Proceeding to merge.

**Pull request merged**